src/qt/bitcoingui.cpp

```diff
@@ -786,8 +801,15 @@ void BitcoinGUI::setNumBlocks(int count, const QDateTime& blockDate, double nVer
             if (header) {
                 return;
             }
-            // Case: not Importing, not Reindexing and no network connection
-            progressBarLabel->setText(tr("No block source available..."));
+            if (clientModel->getNumConnections() == 0)
+            {
+                // Case: not yet connected to peers
+                progressBarLabel->setText(tr("Connecting to peers..."));
+            }
+            else if (!updateHeadersSyncProgressLabel()) {
+                // Case: not Importing, not Reindexing and no network connection
```

Comment:
I think this is dead code, because whenever `getNumConnections()==0`, you display `Connecting ...`. And whenever `getNumConnections()>0`, you are in the case of `BLOCK_SOURCE_NETWORK`.